app/helpers/katello/katello_urls_helper.rb

```diff
@@ -11,5 +11,11 @@ def subscription_manager_configuration_url(host = nil, rpm = true)
 
       "#{prefix}/pub/#{config}"
     end
+
+    def liveimg_url(address, default_schema: 'http', secure: true)
+      result = address
```

Comment:
We should be able to calculate the address based on the `@host.content_source` (see the `subscription_manager_configuration_url` method), and the lifecycle environment the host is assigned to, by using https://github.com/Katello/katello/blob/2934f376473e341ffbb10ac1ffd42efb8c5bdab8/app/models/katello/glue/pulp/repos.rb#L9

So at the end, there should not be `lifveimg_url`, but rather, `repository_url` that takes relative path to the product/repo to be used (such as 'centos7/my_repo') and it generates the full url (that has form of `https://myproxy.example.com/pulp/repos/MyOrg/MyEnv/centos7/my_repo`). One pointer: the `environment` argument that `repo_path_from_content_path` takes is actually `@host.content_facet.lifecycle_environment`)

If you prepare a katello environment with some synchronized repository, I can give you more background on how the whole thing works tomorrow.

Reply:
The signature of the `repository_url` would be:

```
def repository_url(content_path)
end
```

Where content path would be just the last part of the katello repository (such as `centos7/my_repo`)

Reply:
so that what i was missing in understanding, thanks :)